VSRAD.Package/Server/BreakStateDispatchParameters.cs

```diff
@@ -31,9 +30,8 @@ private BreakStateDispatchParameters(uint waveSize, uint gridX, uint gridY, uint
             GridSizeY = Math.Max(1, gridY);
             GridSizeZ = Math.Max(1, gridZ);
             DimX = gridX / GroupSizeX;
-            DimY = gridY > 1 ? gridY / GroupSizeY : 0;
-            DimZ = gridZ > 1 ? gridZ / GroupSizeZ : 0;
-            NDRange3D = gridY > 1 || gridZ > 1;
+            DimY = gridY / GroupSizeY;
+            DimZ = gridZ / GroupSizeZ;
```

Comment:
1. I don't remember why `Math.Max` is needed, but I would change `DimX = gridX / GroupSizeX` to `DimX = GroupSizeX / GroupSizeX`. Same for `DimY`, `DimZ`.
2. Is there any reason for having `Math.Max(1, ...)`?

Reply:
Regarding `Math.Max`, I suspect it's used in the constructor to enforce the invariant that `SizeX/Y/Z >= 1`. Which is not that bad of an idea, considering that we assume it holds true in other parts of the code. I wonder, though, if it would be better to throw an exception instead of silently swallowing invalid values.